theta_dup/include/theta_sketch_dup.h

```diff
@@ -0,0 +1,2209 @@
+      serial_version, theta_sketch_dup_alloc<A>::SERIAL_VERSION);
+  theta_sketch_dup_alloc<A>::check_seed_hash(
+      seed_hash, theta_sketch_dup_alloc<A>::get_seed_hash(seed));
+  return internal_deserialize(is, rf, lg_cur_size, lg_nom_size, flags_byte,
+                              seed);
+}
+
+template <typename A>
+update_theta_sketch_dup_alloc<A>
+update_theta_sketch_dup_alloc<A>::internal_deserialize(
+    std::istream& is, resize_factor rf, uint8_t lg_cur_size,
+    uint8_t lg_nom_size, uint8_t flags_byte, uint64_t seed) {
+  uint32_t num_keys;
+  is.read((char*)&num_keys, sizeof(num_keys));
+  float p;
+  is.read((char*)&p, sizeof(p));
+  uint64_t theta;
+  is.read((char*)&theta, sizeof(theta));
+  vector_u64<A> keys(1 << lg_cur_size);
+  is.read((char*)keys.data(),
+          sizeof(std::pair<uint64_t, uint64_t>) * keys.size());
+  const bool is_empty =
+      flags_byte & (1 << theta_sketch_dup_alloc<A>::flags::IS_EMPTY);
+  if (!is.good()) throw std::runtime_error("error reading from std::istream");
+  return update_theta_sketch_dup_alloc<A>(is_empty, theta, lg_cur_size,
+                                          lg_nom_size, std::move(keys),
+                                          num_keys, rf, p, seed);
+}
+
+template <typename A>
+update_theta_sketch_dup_alloc<A> update_theta_sketch_dup_alloc<A>::deserialize(
+    datasketches_pb::Update_theta_sketch_dup& pb, uint64_t seed) {
+  uint8_t serial_version = pb.serial_version();
+  uint8_t type = pb.sketch_type();
+  resize_factor rf = static_cast<resize_factor>(pb.rf());
+  uint8_t lg_nom_size = pb.lg_nom_size();
+  uint8_t lg_cur_size = pb.lg_cur_size();
+  uint8_t flags_byte = pb.flags_byte();
+  uint16_t seed_hash = pb.seed_hash();
+  theta_sketch_dup_alloc<A>::check_sketch_type(type, SKETCH_TYPE);
+  theta_sketch_dup_alloc<A>::check_serial_version(
+      serial_version, theta_sketch_dup_alloc<A>::SERIAL_VERSION);
+  theta_sketch_dup_alloc<A>::check_seed_hash(
+      seed_hash, theta_sketch_dup_alloc<A>::get_seed_hash(seed));
+  return internal_deserialize(pb, rf, lg_cur_size, lg_nom_size, flags_byte,
+                              seed);
+}
+
+template <typename A>
+update_theta_sketch_dup_alloc<A>
+update_theta_sketch_dup_alloc<A>::internal_deserialize(
+    datasketches_pb::Update_theta_sketch_dup& pb, resize_factor rf,
+    uint8_t lg_cur_size, uint8_t lg_nom_size, uint8_t flags_byte,
+    uint64_t seed) {
+  uint32_t num_keys = pb.num_keys();
+  float p = pb.p();
+  uint64_t theta = pb.theta();
+  vector_u64<A> keys(1 << lg_cur_size);
+  for (int i = 0; i < pb.keys_size(); i++)
+    keys[i] = std::make_pair(pb.keys(i).hash_val(), pb.keys(i).cnt());
+  const bool is_empty =
+      flags_byte & (1 << theta_sketch_dup_alloc<A>::flags::IS_EMPTY);
+  return update_theta_sketch_dup_alloc<A>(is_empty, theta, lg_cur_size,
+                                          lg_nom_size, std::move(keys),
+                                          num_keys, rf, p, seed);
+}
```

Comment:
Definition: derived class update_theta_sketch_dup_alloc, deserialize (and internal_deserialize) using protobuf.